drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/renderers/ScenarioGridColumnRenderer.java

```diff
@@ -109,12 +110,16 @@ protected void applyBackgroundColor(ScenarioGridCell cell,
         }
     }
 
-    protected String getValueToShow(ScenarioGridCell scenarioGridCell){
+    protected String getValueToShow(ScenarioGridCell scenarioGridCell) {
         String rawValue = scenarioGridCell.getValue() != null ? scenarioGridCell.getValue().getValue() : null;
-        return (rawValue != null && (scenarioGridCell.isList() || scenarioGridCell.isMap())) ? getCollectionString(rawValue, scenarioGridCell.isList()) : rawValue;
+        return (rawValue != null && (scenarioGridCell.isList() || scenarioGridCell.isMap())) ? getCollectionString(rawValue, scenarioGridCell.isList(), scenarioGridCell.isExpression()) : rawValue;
     }
 
-    protected String getCollectionString(String jsonString, boolean isList) {
+    protected String getCollectionString(String jsonString, boolean isList, boolean isExpression) {
+        if (isExpression) {
+            String toFormat = isList ? "List() %s" : "Map() %s";
+            return toFormat.replace("%s", ConstantHolder.EXPRESSION);
```

Comment:
Not sure about replace performance, but it is the same as:
```java
            String toFormat = isList ? "List() " : "Map() ";
            return toFormat + ConstantHolder.EXPRESSION;
```

Reply:
Done